core/arch/arm/plat-ls/ls_i2c_test.c

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 NXP
+ * Test suite for I2C Controller
+ */
+
+#include <drivers/ls_i2c.h>
+#include <initcall.h>
+#include <kernel/delay.h>
+#include <kernel/tee_time.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+#include <stddef.h>
+
+#define PCF2129_CTRL3_BIT_BLF  BIT(2) /* Battery Low Flag*/
+#define PCF2129_SLAVE_ADDRESS  0x51
+
+struct __packed pcf2129_regs {
+	uint8_t control[3];
+	uint8_t seconds;
+	uint8_t minutes;
+	uint8_t hours;
+	uint8_t days;
+	uint8_t weekdays;
+	uint8_t months;
+	uint8_t years;
+	uint8_t second_alarm;
+	uint8_t minute_alarm;
+	uint8_t hour_alarm;
+	uint8_t day_alarm;
+	uint8_t weekday_alarm;
+};
+
+/* I2c clock based on 750Mhz platform clock */
+#define I2C_CLOCK 93750000
+#define I2C_SPEED 100000
+
+static TEE_Result i2c_test_suite(void)
+{
+	struct ls_i2c_data i2c_data = {0};
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct i2c_reg_request req = {0};
+	uint8_t rtc_reg_adr = 0;
+	uint8_t retries = 5;
+	struct pcf2129_regs *pcf_regs = (struct pcf2129_regs *)malloc
+					(sizeof(struct pcf2129_regs));
+
+	EMSG("I2C RTC TEST: will get time from RTC 5 times after 2 secs");
+
+	/* set slave info */
+	i2c_data.i2c_controller = CFG_LS_I2C_CONTROLLER;
+	i2c_data.i2c_bus_clock = I2C_CLOCK;
+	i2c_data.speed = I2C_SPEED;
+
+	/* Initialise DSPI driver */
+	res = i2c_init(&i2c_data);
+	if (res) {
+		EMSG("Unable to init I2C driver");
+		goto exit;
+	}
+
+	while (retries--) {
+		rtc_reg_adr = offsetof(struct pcf2129_regs, control[2]);
+
+		req.operation_count = 1;
+		req.operation[0].flags = 0;
```

Comment:
Quick question : The write operation flag is = 0 ?

Reply:
Yes, 0 is for WRITE, i will define a #define for that also.